protocol/binary_test.go

```diff
@@ -284,6 +285,18 @@ func TestBinary(t *testing.T) {
 	checkEncodeDecode(t, wire.TBinary, tests)
 }
 
+func TestBinaryLargeLength(t *testing.T) {
+	// 5 MB + 4 bytes for length
+	data := make([]byte, 5242880+4)
+	data[0], data[1], data[2], data[3] = 0x0, 0x50, 0x0, 0x0 // 5 MB
+
+	value, err := Binary.Decode(bytes.NewReader(data), wire.TBinary)
+	require.NoError(t, err, "failed to decode value")
+
+	want := wire.NewValueBinary(data[4:])
+	assert.True(t, wire.ValuesAreEqual(want, value), "values did not match")
+}
```

Comment:
this test will pass with or without this change right? i would have preferred a test where we have a small buffer but where the length is invalid (max size). However with 4 bytes, the length could be 4gb which may succeed with a make.